JSONObject.java

```diff
@@ -578,10 +581,10 @@ public double getDouble(String key) throws JSONException {
         Object object = this.get(key);
         try {
             return object instanceof Number ? ((Number) object).doubleValue()
-                    : Double.parseDouble((String) object);
+                    : new BigDecimal((String) object).doubleValue();
```

Comment:
I doubt that, things are quite clear in the Java Specification, anyone that writes a JVM must follow these rules, and most likely will use native calls:

`public static native double longBitsToDouble(long bits);`

I recommended that for `double` and `float`, it is an unnecessary instantiation.

as for `int` and `long` there are waaaaay faster and cheap ways:
- `(int)Double.parseDouble(x)`
- `(long)Double.parseDouble(x)`